### PR TITLE
chore: Bump Rust version to 1.78 in rust-toolchain

### PR DESCRIPTION
**Pull request by frankdavid**
A transitive dependency (cargo-platform) has been updated and now requires min Rust version 1.78.

Our library can still be used with a lower Rust version but our tests/benchmarks will be run with 1.78.

### CODE REVIEW

**Review by berestovskyy — APPROVED** [review]
Approving to unblock you. We should figure out why, for example, `vec_insert_u64` regresses ~5%...

**Comment by frankdavid** [comment]
> We should figure out why, for example, vec_insert_u64 regresses ~5%...

I think there are two things at play here:
- Rust optimizes overall performance, individual benchmarks may get slower or faster depending on code generation.
- This effect is amplified even more in our case because we have our own way of measuring performance via instruction count and custom weighting which sometimes doesn't correspond to real world performance (for example, it doesn't take into account CPU caches and branch prediction, furthermore function call overhead is discounted). As I was exploring places in the code that could be optimized, it happened that a change that I expected to greatly improve performance, actually increased the instruction count. I can imagine that there are changes in the Rust stdlib which improved the metrics that the stdlib maintainers optimize for but made things worst according to our metrics.

**Pull request merged**
